metpy/calc/thermo.py

```diff
@@ -230,8 +230,15 @@ def lfc(pressure, temperature, dewpt):
     # that point to get the real first intersection for the LFC calculation.
     x, y = find_intersections(pressure[1:], ideal_profile[1:], temperature[1:],
                               direction='increasing')
+    # Two possible cases here: LFC = LCL, or LFC doesn't exist
     if len(x) == 0:
-        return np.nan * pressure.units, np.nan * temperature.units
+        if np.any(ideal_profile.magnitude > temperature.magnitude) > 0.:
```

Comment:
You shouldn't need the `> 0.`; `np.any()` will return `True/False`. Also, I the `.magnitude` isn't necessary, and I'm not sure it's guaranteed correct (i.e. I'm not sure it's true that `ideal_profile` and `temperature` are always the same units. So this line should be:
```python
if np.any(ideal_profile > temperature):
```